### DISCUSSION TITLE
How to set up a Roche-lobe filling star?

### DISCUSSION THREAD

**Discussion by AstroJLin** [phoebe-project/phoebe2#577] [General]
Hi,

I want to know how to set up a Roche-lobe filling star in a semidetached binary system ?

Thanks!

**Comment by kecnry** [comment]
You'll need to constrain the radius as in the [semi-detached constraint tutorial](http://phoebe-project.org/docs/latest/tutorials/requiv_crit_semidetached).  Let us know if you run into any questions or issues.

**Comment by AstroJLin** [comment]
Hi,
Thank you for your reply!

You means that  I need to set up b.add_constraint('semidetached', 'primary'), which will result in not allow to set up the initial radius of a primary in the model？ In addition, when I set up b.add_constraint('semidetached', 'primary'), which condition not only operate in current model but also apply for the parameters given by the mcmc (i.e. The post parameters given by the mcmc  is obtained within a Roche-lobe filling star)?

**Reply by kecnry** [reply]
Yes, that's right.  The constraint will update the radius whenever any parameter that affects it is changed and will no longer be able to be changed directly/manually, including within fitting.  In order to access the posterior distribution on the radius, you'll need to [propagate the distribution through the constraint](http://phoebe-project.org/docs/latest/tutorials/distribution_propagation).
